src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```
@@ -1196,5 +1199,224 @@ internal static bool TryGetInt(FormulaValue value, out int outputValue)
+                    DateValue dv => WithQuotes(TimeZoneInfo.ConvertTimeToUtc(dv.Value, _timeZoneInfo).ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)),
+                    DecimalValue dmv => dmv.Value.ToString(CultureInfo.InvariantCulture),
+                    GuidValue gv => WithQuotes(gv.Value.ToString("D", CultureInfo.InvariantCulture)),
+                    NumberValue nv => nv.Value.ToString(CultureInfo.InvariantCulture),
+                    OptionSetValue osv => GetOptionSetValueString(osv),
+                    RecordValue rv => GetJsonStringForRecord(rv, indent),
+                    StringValue sv => WithQuotes(sv.Value),
+                    TableValue tv => GetJsonStringForTable(tv, indent),
+                    TimeValue tmv => WithQuotes(tmv.Value.ToString(@"hh\:mm\:ss\.fff", CultureInfo.InvariantCulture)),
+                    _ => "invalid",
+                };
+            }
+
+#pragma warning restore CS0618
+
+            private static string WithQuotes(string str) => $@"""{str}""";
+
+            internal static string GetColorString(Color color) => WithQuotes($"#{color.R:x2}{color.G:x2}{color.B:x2}{color.A:x2}");
+
+            internal static string GetOptionSetValueString(OptionSetValue optionSetValue)
+            {
+                if (optionSetValue.Type._type.OptionSetInfo is EnumSymbol enumSymbol)
+                {
+                    if (enumSymbol.EntityName == "Color")
+                    {
+                        Color color = Color.FromArgb((int)(uint)(double)optionSetValue.ExecutionValue);
+                        return GetColorString(color);
+                    }
+
+                    return WithQuotes(optionSetValue.ExecutionValue.ToString());
+                }
+                else if (optionSetValue.Type._type.OptionSetInfo is IExternalOptionSet ops)
+                {
+                    return WithQuotes(optionSetValue.Option);
+                }
+
+                return null;

```

>return null; [](http://example.com/codeflow?start=16&length=12)

this looks concerning, do we append null in SB in some case?

---

this should really throw an exception, i'll fix that